lib/API/VK/Device.cpp

```diff
@@ -1378,9 +1408,18 @@ class VKDevice : public offloadtest::Device {
       llvm::SmallVector<char> SpecData;
       VkSpecializationInfo SpecInfo = {};
       if (!Shader.SpecializationConstants.empty()) {
+        llvm::DenseSet<uint32_t> SeenConstantIDs;
         for (const auto &SpecConst : Shader.SpecializationConstants) {
+          if (!SeenConstantIDs.insert(SpecConst.ConstantID).second)
+            return llvm::createStringError(
+                std::errc::invalid_argument,
+                "Specialization constant ID %u is already defined",
```

Comment:
nit: would change the error to something more explicit about the location of the issue:
"YAML/config contains conflicting entries for constant ID %u"

(Looking at the test, the error could be interpreted as "we disallow constant ID reuse in the shader itself")